ectoolkits/analysis/disdeg.py

```diff
@@ -0,0 +1,181 @@
+import os
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+from MDAnalysis.lib.distances import capped_distance
+from MDAnalysis.analysis.base import AnalysisBase
+
+
+
+def count_Oidxs_bonds(positions, o_idxs:list, conf_idxs, box, max_cutoff=1.4, min_cutoff=None, ):
+    
+    _res = capped_distance(positions[o_idxs], 
+                           positions[conf_idxs], 
+                           box=box, min_cutoff=min_cutoff, max_cutoff=max_cutoff,
+                          return_distances=False)
+    #print(_res)
+    _res = _res[:,0]
+    
+    # _res dont care the H idx
+    # conf_idxs = 
+    #TODO: save the H idxs?
+    return np.bincount(_res, minlength=len(o_idxs))
+
+def _count_disdeg(arr):
+    return np.count_nonzero(arr < 2)/len(arr)
+    
+def _count_percent_o(arr):
+    return np.count_nonzero(arr == 0)/len(arr)
+
+def _count_percent_oh(arr):
+    return np.count_nonzero(arr == 1)/len(arr)
+
+def _count_percent_h2o(arr):
+    return np.count_nonzero(arr == 2)/len(arr)
+
+def _count_percent_h3o(arr):
+    return np.count_nonzero(arr == 3)/len(arr)
+
+def cumsum_arr(arr):
+    #not vectorize:
+    #seem same in ectoolkits utils
+    cumulative_sum = np.cumsum(arr)
+    for i,_num in enumerate(cumulative_sum):
+        cumulative_sum[i] =cumulative_sum[i]/(i+1)
+    return cumulative_sum
+
+class DisDeg(AnalysisBase):
+    
+    
+    def __init__(self,
+                 atomgroup,
+                 terminal_oxygen_idxs,
```

Comment:
How about support two flavors of idxs. Here I assume N is the number of terminal oxygen idxs
1. users provide terminal oxygen idx with shape of (N, ). In your code, the idxs remain unchanged for whole trajectory.
2. users provide terminal oxygen idx with shape of (num_frames, N ). This allows the idxs of terminal oxygen to be updated.